CMakeLists.txt

```diff
@@ -0,0 +1,340 @@
+
+cmake_minimum_required(VERSION 3.4.3)
+
+list(APPEND CMAKE_MODULE_PATH
+     "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+
+project(Foundation
+        LANGUAGES
+          C)
+
+option(FOUNDATION_ENABLE_LIBDISPATCH "Enable GCD Support" YES)
+option(FOUNDATION_PATH_TO_LIBDISPATCH_SOURCE "Path to libdispatch source" "")
+option(FOUNDATION_PATH_TO_LIBDISPATCH_BUILD "Path to libdispatch build" "")
+
+find_package(CURL REQUIRED)
+find_package(ICU COMPONENTS uc i18n REQUIRED)
+find_package(LibXml2 REQUIRED)
+
+include(SwiftSupport)
+include(GNUInstallDirs)
+include(ExternalProject)
+
+ExternalProject_Add(CoreFoundation
+                    SOURCE_DIR
+                      ${CMAKE_CURRENT_SOURCE_DIR}/CoreFoundation
+                    CMAKE_COMMAND
+                      ${CMAKE_COMMAND}
+                    CMAKE_ARGS
+                      -DBUILD_SHARED_LIBS=NO
+                      -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                      -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+                      -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+                      -DCMAKE_INSTALL_LIBDIR=usr/lib
```

Comment:
The "lib" part really should be a variable, like "SWIFT_LIBDIR", or something like that, with a default "lib" value. That way, Linux package maintainers could define it in build time, and it will handle the whole lib|lib64 debacle themselves.

Something like that before the options on this file:
set(SWIFT_LIBDIR "lib" CACHE STRING "Library folder name, defined by swift main buildscript")

CMake [do have a way to handle this by itself](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html), with the GNUInstallDirs module, and that maybe a ever more elegant solution where you can use the standard CMAKE_INSTALL_LIBDIR generated by it and remove this manual definition from you file, but i don't know how it behaves on OS X. In theory it should revert to plain "lib".

Reply:
@capent - the install is already using GNUInstallDirs.  This is for the internal build of CoreFoundation for Foundation.  This allows us to simplify some of the logic in the build by assuming it is always lib.